rhizome/host/bin/convert-encrypted-dek-to-vhost-backend-conf

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require "json"
+require "yaml"
+require "optparse"
+require "openssl"
+require "base64"
+
+def validate_encrypted_dek_file(data)
+  unless data.key?("cipher") && data["cipher"].is_a?(String)
+    raise ArgumentError, "DEK file must contain 'cipher' field as a string"
+  end
+  unless data.key?("key") && data["key"].is_a?(Array) && data["key"].length == 2 && data["key"].all?(String)
+    raise ArgumentError, "DEK file must contain 'key' field as an array with exactly two strings"
+  end
+  unless data.key?("key2") && data["key2"].is_a?(Array) && data["key2"].length == 2 && data["key2"].all?(String)
+    raise ArgumentError, "DEK file must contain 'key2' field as an array with exactly two strings"
+  end
+end
+
+def validate_kek_file(data)
+  unless data.is_a?(Hash)
+    raise ArgumentError, "KEK file must be a dictionary"
+  end
+  data.each do |key, value|
+    unless key.is_a?(String)
+      raise ArgumentError, "KEK file keys must be strings"
+    end
+    unless value.is_a?(Hash)
+      raise ArgumentError, "KEK file values must be dictionaries"
+    end
+    required_fields = ["key", "init_vector", "auth_data", "algorithm"]
+    missing_fields = required_fields - value.keys
+    unless missing_fields.empty?
+      raise ArgumentError, "KEK entry '#{key}' is missing required fields: #{missing_fields.join(", ")}"
+    end
+    required_fields.each do |field|
+      unless value[field].is_a?(String)
+        raise ArgumentError, "KEK entry '#{key}' field '#{field}' must be a string"
+      end
+    end
+  end
+end
+
+def unwrap_key(encrypted_key, kek_data)
+  algorithm = kek_data["algorithm"]
+  fail "currently only aes-256-gcm is supported" unless algorithm == "aes-256-gcm"
+
+  decipher = OpenSSL::Cipher.new(algorithm)
+  decipher.decrypt
+  decipher.key = Base64.decode64(kek_data["key"])
+  decipher.iv = Base64.decode64(kek_data["init_vector"])
+  decipher.auth_data = kek_data["auth_data"]
+
+  auth_tag = encrypted_key[1]
+  decoded_auth_tag = Base64.decode64(auth_tag)
+
+  # We reject if auth_tag length is not within acceptable range for GCM
+  fail "Invalid auth_tag size: #{decoded_auth_tag.bytesize}" unless (12..24).cover(decoded_auth_tag.bytesize)
```

Comment:
I see this was fixed in a later commit, but in general it's better to use git rebase so it doesn't stick out like an obvious issue when doing a commit-by-commit review.